backend-shared/util/operations/types_test.go

```diff
@@ -102,6 +103,188 @@ var _ = Describe("Testing CreateOperation function.", func() {
 			Expect(dbOperationFirst.SeqID).To(Equal(dbOperationSecond.SeqID))
 		})
 	})
+
+	Context("Testing waitForOperationToComplete and IsOperationComplete function.", func() {
+		var ctx context.Context
+		var dbq db.AllDatabaseQueries
+
+		delayUpdateOfState := func(operation *db.Operation, k8sClient client.Client, gitopsEngineInstance db.GitopsEngineInstance) {
+			Eventually(func() bool {
+				// Allow waitForOperationToComplete to iterate/wait
+				time.Sleep(2 * time.Second)
+
+				operationget := db.Operation{
+					Operation_id: operation.Operation_id,
+				}
+
+				err := dbq.GetOperationById(ctx, &operationget)
+				// The operation might not be created yet
```

Comment:
I think we can add Expect statement here `Expect(err).To(HaveOccurred())`, I am not sure, please let me know if its not required

Reply:
I didn't want the test to fail if it isn't created yet, so it will eventually be true.